src/webgpu/shader/validation/shader_io/group_and_binding.spec.ts

```diff
@@ -0,0 +1,116 @@
+export const description = `Validation tests for group and binding`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { ShaderValidationTest } from '../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('storage')
+  .desc(`Test validation of group and binding on storage resources`)
+  .fn(t => {
+    const code = `
+@group(1) @binding(1)
+var<storage> a: i32;
+
+@workgroup_size(1, 1, 1)
+@compute fn main() {
+  _ = a;
+}`;
+    t.expectCompileResult(true, code);
+  });
+
+g.test('uniform')
+  .desc(`Test validation of group and binding on uniform resources`)
+  .fn(t => {
+    const code = `
+@group(1) @binding(1)
+var<uniform> a: i32;
+
+@workgroup_size(1, 1, 1)
+@compute fn main() {
+  _ = a;
+}`;
+    t.expectCompileResult(true, code);
+  });
+
+g.test('texture')
+  .desc(`Test validation of group and binding on texture resources`)
+  .fn(t => {
+    const code = `
+@group(1) @binding(1)
+var a: texture_2d<f32>;
+
+@workgroup_size(1, 1, 1)
+@compute fn main() {
+  _ = a;
+}`;
+    t.expectCompileResult(true, code);
+  });
+
+g.test('sampler')
+  .desc(`Test validation of group and binding on sampler resources`)
+  .fn(t => {
+    const code = `
+@group(1) @binding(1)
+var a: sampler;
+
+@group(0) @binding(2)
+var b: sampler_comparison;
+
+@workgroup_size(1, 1, 1)
+@compute fn main() {
+  _ = a;
+  _ = b;
+}`;
+    t.expectCompileResult(true, code);
+  });
+
```

Comment:
Negative tests for variables that need a binding point which don't have one?

Reply:
done